redfish-core/include/utils/chassis_utils.hpp

```diff
@@ -61,5 +61,258 @@ void getValidChassisID(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+}
+
+template <typename Callback>
+inline void checkAssociation(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const std::string& chassisPath,
+                             Callback&& callback)
+{
+    BMCWEB_LOG_DEBUG << "Check chassis for association";
+
+    std::string chassis =
+        sdbusplus::message::object_path(chassisPath).filename();
+    if (chassis.empty())
+    {
+        BMCWEB_LOG_ERROR << "Failed to find / in Chassis path";
+        messages::internalError(aResp->res);
+        return;
+    }
+
+    // check if this chassis hosts any association
+    crow::connections::systemBus->async_method_call(
+        [aResp, chassisPath, callback{std::move(callback)}](
+            const boost::system::error_code ec,
+            const std::vector<std::pair<std::string, std::vector<std::string>>>&
+                object) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            for (const auto& [serviceName, interfaceList] : object)
+            {
+                for (auto& interface : interfaceList)
+                {
+                    if (interface ==
+                        "xyz.openbmc_project.Association.Definitions")
+                    {
+                        checkForAssemblyAssociations(aResp, chassisPath,
+                                                     serviceName,
+                                                     std::move(callback));
+                        return;
+                    }
+                }
+            }
+            return;
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject", chassisPath,
+        std::array<const char*, 0>{});
+}
+
+template <typename Callback>
+inline void getChassisAssempbly(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                                Callback&& callback)
+{
+    BMCWEB_LOG_DEBUG << "Get chassis path";
+
+    // get the chassis path
+    crow::connections::systemBus->async_method_call(
+        [aResp, callback{std::move(callback)}](
+            const boost::system::error_code ec,
+            const std::vector<std::string>& chassisPaths) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            // check if the chassis path belongs to the chassis ID passed
+            for (const auto& path : chassisPaths)
+            {
+                BMCWEB_LOG_DEBUG << "Chassis Paths from Mapper " << path;
+                std::string chassis =
+                    sdbusplus::message::object_path(path).filename();
+                if (chassis != "chassis")
```

Comment:
Chassis could be "chassis" or "chassis15363". I believe the later one is MEX/IO module or something. The assembly numbers/IDs are generated for each of chassis type w/ starting number 0. We may be good for now as we are only interested in wiliwaks/disk backplane assembly IDs. But if there is a need to get the assembly numbers for MEX/IO module, it is going to fail. I'm not sure whether we need to think about MEX assemblies or not though. 

@gtmills it's your call whether we need make it generic enough to handle all supported chassis or just stick with "chassis"

Reply:
@sbteeks make it generic, not needed now, but let me know what you needed

Reply:
It is for @gtmills to decide whether there will be a need in the future to get assembly ID from other chassis - like chassis15363 as an example. For our current needs just "chassis" type is good.